src/entrypoint.cpp

```diff
@@ -271,6 +308,76 @@ namespace aperf {
                                                    event_name));
       }
 
+      for (std::string &metric_str : metric_strs) {
+        std::vector<std::string> matches_metric = tokenize(metric_str, ',');
+
+        SubcommandData metric_command_data = {"", "", 0, ""};
+
+        for (std::string &option : matches_metric){
+            std::vector<std::string> parsed_option = tokenize(option, ':');
+            if(parsed_option.size() == 2){
+                if(parsed_option[0] == "c")
+                if(metric_command_data.metric_command.empty()){
+                    metric_command_data.metric_command = parsed_option[1];
+                }else{
+                  print("Cannot specify multiple metric commands", true, true);
+                  return 2;
+                }
+                else if(parsed_option[0] == "n")
+                if(metric_command_data.metric_name.empty()){
+                    metric_command_data.metric_name = parsed_option[1];
+                }else{
+                  print("Cannot specify multiple metric names", true, true);
+                  return 2;
+                }
+                else if(parsed_option[0] == "f")
+                if(metric_command_data.frequency == 0){
+                try {
+                  metric_command_data.frequency = std::stoi(parsed_option[1]);
+                  if (metric_command_data.frequency < 0) {
```

Comment:
What if the user specifies one or more frequencies equal to 0 (which is not correct)?

Reply:
Added check to not allow for frequency value to be 0.